lstchain/tools/lstchain_create_drs4_pedestal_file.py

```diff
@@ -0,0 +1,112 @@
+"""
+Create drs4 time pedestal fits file.
+"""
+import numpy as np
+from astropy.io import fits
+from tqdm.autonotebook import tqdm
+
+from ctapipe.core import Provenance, traits
+from ctapipe.core import Tool
+from ctapipe.io import EventSource
+from lstchain.calib.camera.drs4 import DragonPedestal
+
+
+class PedestalFITSWriter(Tool):
+
+    name = "PedestalFITSWriter"
+    description = "Generate a pedestal FITS file"
+
+    output = traits.Path(
+        default_value="pedestal.fits",
+        help="Path to the generated fits pedestal file",
+        directory_ok=False,
+    ).tag(config=True)
+
+    deltaT = traits.Bool(
+        help="Use delta T correction", default_value=False
+    ).tag(config=True)
+
+    progress_bar = traits.Bool(
+        help="show progress bar during processing",
+        default_value=True,
+    ).tag(config=True)
+
+    aliases = {
+        ("i", "input"): "EventSource.input_url",
+        ("o", "output"): "PedestalFITSWriter.output",
+        "max-events": "EventSource.max_events",
+        "start-r0-waveform": "DragonPedestal.r0_sample_start",
+    }
+    flags = {
+        "deltaT": (
+            {"PedestalFITSWriter": {"deltaT": True}},
+            "Activate delta T corrections",
+        )
+    }
+
+    classes = [EventSource, DragonPedestal]
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        """
+        Tool that generates a pedestal FITS file for low level calibration.
+
+        For getting help run:
+        lstchain_create_drs4_pedestal_file --help
+        """
+
+        self.eventsource = None
+        self.pixel_ids = None
+        self.pedestal = None
+
+    def setup(self):
+
+        self.log.debug("Opening file")
+        self.eventsource = EventSource.from_config(parent=self)
+
+    def start(self):
+
+        event = next(iter(self.eventsource))
+        tel_id = event.trigger.tels_with_trigger[0]
+        self.pixel_ids = event.lst.tel[tel_id].svc.pixel_ids
+        self.pedestal = DragonPedestal(tel_id=tel_id, n_module=event.lst.tel[tel_id].svc.num_modules, config=self.config)
```

Comment:
`DragonPedestal` is a component. So please do not pass `config=self.config` but `parent=self`.

Reply:
You should probably also already create it in `setup`. that's what setup is for.

Reply:
And add it to `classes`